token/cli/src/config.rs

```diff
@@ -250,14 +256,54 @@ impl<'a> Config<'a> {
         }
     }
 
+    // Returns Ok(default signer), or Err if there is no default signer configured
+    pub(crate) fn default_signer(&self) -> Result<Arc<dyn Signer>, Error>
+    {
+        if let Some(default_signer) = &self.default_signer {
+            Ok(default_signer.clone())
+        }
+        else {
+            Err("default signer is required, please specify a valid default signer by identifying a \
+                 valid configuration file using the --config-file argument, or by creating a valid \
+                 config at the default location of ~/.config/solana/cli/config.yml using the solana \
+                 config command".to_string().into())
+        }
+    }
+    
+    // Returns Ok(default signer pubkey), or Err if there is no default signer configured
+    pub(crate) fn default_signer_pubkey(&self) -> Result<Pubkey, Error>
+    {
+        self.default_signer().map(|a| a.pubkey().clone())
+    }
```

Comment:
nit: This function makes things a bit more confusing to me, since you can just as easily do `config.default_signer()?.pubkey()`.  If you do really want to keep it though, we should be able to get by without the `clone()` call since `pubkey()` creates a new one

Reply:
> nit: This function makes things a bit more confusing to me, since you can just as easily do `config.default_signer()?.pubkey()`. If you do really want to keep it though, we should be able to get by without the `clone()` call since `pubkey()` creates a new one

Yeah my lack of Rust experience shows through.  I will change to what you suggest.